lib/cog/command/pipeline/executor.ex

```diff
@@ -677,35 +663,160 @@ defmodule Cog.Command.Pipeline.Executor do
   ########################################################################
   # Error Handling Functions
 
-  defp send_error(error, state) when is_binary(error),
-    do: send_reply("Whoops! An error occurred. #{error}", state)
-  defp send_error(error, state) do
-    Logger.warn("The error message #{inspect error} should be in string format for displaying to the user")
-    send_reply("Whoops! An error occurred. #{inspect error}", state)
-  end
+  # Create a standard error message for pipeline-related
+  # errors. Eventually this will be templated.
+  defp user_error(error, state) do
+    id = state.id
+    started = Cog.Events.Util.ts_iso8601_utc(state.started)
+    initiator = sender_mention_name(state)
+    pipeline_text = state.request["text"]
+    error_message = format_user_error(error, state)
+    failing = case state.current_plan do
```

Comment:
Instead of doing this dance to figure out what failed during execution, would it be worth storing `failing` in `state`? Not sure if that's a great idea or not, but it does seem like it would clean up a bit of this logic.